### PR TITLE
[1Password] Add 1Password Events API Audit Events

### PR DESCRIPTION
**Pull request by anthonythleung**
- Enhancement

## What does this PR do?

This PR adds a new audit events data stream to the 1Password Elastic integration.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Screenshots

image

### CODE REVIEW

**Comment by efd6** [comment]
Can you provide a link to the auditevents API and describe the provenance of the test cases (constructed/sanitised/etc).

File: packages/1password/kibana/dashboard/1password-audit-events-full-dashboard.json [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+            "id": "1password-audit-events"
+        },
+        {
+            "type": "search",
+            "name": "60da356b-c843-4d41-8bf4-04e04ef77734:search_0",
+            "id": "1password-audit-events"
+        }
+    ],
+    "migrationVersion": {
+        "dashboard": "8.6.0"
```
Comment (efd6) [inline packages/1password/kibana/dashboard/1password-audit-events-full-dashboard.json:480]: This means you'll need to bump the kibana version.

**Comment by anthonythleung** [comment]
> Can you provide a link to the auditevents API and describe the provenance of the test cases (constructed/sanitised/etc).

Thanks for the review! The events API audit events data streaming has not been released yet, so I don't have any links or documentations for this right now, but the API is scheduled to be released on Mar 15, so I'll have the documentations available by then. The test data were created with a local development build of the Events API with local test data.

**Comment by anthonythleung** [comment]
@efd6 The audit events API has now been released and the documentations are available here: https://developer.1password.com/docs/events-api/reference/#post-apiv1auditevents

**Comment by efd6** [comment]
The agent is complaining:
```
2023-03-16 09:45:11 {"log.level":"error","@timestamp":"2023-03-15T23:15:11.961Z","message":"Error while processing http request: failed to execute rf.collectResponse: failed to execute http client.Do: server responded with status code 404: ","component":{"binary":"filebeat","dataset":"elastic_agent.filebeat","id":"httpjson-default","type":"httpjson"},"log":{"source":"httpjson-default"},"ecs.version":"1.6.0","log.logger":"input.httpjson-cursor","log.origin":{"file.line":135,"file.name":"httpjson/input.go"},"id":"httpjson-1password.audit_events-23c79f8b-96db-46c4-a472-ae89233f01ff","input_source":"http://elastic-package-service_1password_eventsapi_mock_1:8080/api/v1/auditevents","service.name":"filebeat","input_url":"http://elastic-package-service_1password_eventsapi_mock_1:8080/api/v1/auditevents","ecs.version":"1.6.0"}
```
It looks like you are missing an endpoint addition to the docker/config.yml.

**Comment by efd6** [comment]
Sorry, a conflicting change has been made. Would you please resolve the conflict. Ping me when it's done so I won't miss a window. It all looks good to me after that is done.

File: packages/1password/data_stream/audit_events/_dev/test/system/test-default-config.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+input: httpjson
+service: 1password_eventsapi_mock
+vars:
+  url: http://{{Hostname}}:{{Port}}
+  token: --token--
+  preserve_original_event: true
+data_stream:
+  vars: ~
```
Comment (bhapas) [inline packages/1password/data_stream/audit_events/_dev/test/system/test-default-config.yml:8]: There is a new capability in `System Testing` for `httpjson` type input to assert on the number of events ingested in whole after pagination and/or cursors. May be it is worth using it

This helps to catch errors in future if some events get lost due to some changes in the pipeline etc.,

Reference [here](https://github.com/elastic/integrations/blob/main/packages/google_workspace/data_stream/admin/_dev/test/system/test-default-config.yml#L23-L24)

Reference PR - #5326 

```suggestion
  vars: ~
 assert:
   hit_count : xxx
```

File: packages/1password/manifest.yml [hunk 3]
```diff
@@ -10,7 +10,7 @@ categories:
   - credential_management
 release: ga
 conditions:
-  kibana.version: ^8.1.0
+  kibana.version: ^8.6.0
```
[the remark this answers is not in the document]
Comment (P1llus) [reply packages/1password/manifest.yml:13]: I think for this usecase its fine, as we will be bumping the versions quite soon for new dashboard features as well (I have a draft for new 1Password changes that I am working on already).

**Comment by anthonythleung** [comment]
@efd6 I've resolved the merge conflicts and addressed the comments above. Thanks!

**Pull request merged**